### DISCUSSION TITLE
Error: DeviceRole matching query does not exist.

### DISCUSSION THREAD

**Discussion by andytruett** [mattieserver/netbox-topology-views#260] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/7491207/218836347-94e37a9c-fac9-49cf-bd1a-644ef9a4d9be.png)

Getting the above error when I hit Topology Views > Topology Images.

I've tried a handful of things, including clearing my browser cache, but to no avail.

Thanks for any help anyone can give!

**Comment by mattieserver** [comment, marked as answer]
The default config includes the following:

        "preselected_device_roles": [
             "Firewall",
             "Router",
             "Distribution Switch",
             "Core Switch",
             "Internal Switch",
             "Access Switch",
             "Server",
             "Storage",
             "Backup",
             "Wireless AP",
         ],

Maybe you can override this with only the roles that are in use?
I will have to test if this is actually the issue.
________________________________
From: andytruett ***@***.***>
Sent: Tuesday, February 14, 2023 8:24:53 PM
To: mattieserver/netbox-topology-views ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [mattieserver/netbox-topology-views] Error: DeviceRole matching query does not exist. (Discussion #260)


[image]<https://user-images.githubusercontent.com/7491207/218836347-94e37a9c-fac9-49cf-bd1a-644ef9a4d9be.png>

Getting the above error when I hit Topology Views > Topology Images.

I've tried a handful of things, including clearing my browser cache, but to no avail.

Thanks for any help anyone can give!

**Reply by andytruett** [reply]
I must have deleted a device role that had an image assignment within the plugin.

Went to psql > netbox_topology_views_roleimage and deleted the row containing that object_id, and I'm back on track.

Thanks for the plugin and the reply!

**Reply by mroupie34** [reply]
Hello Teams,

I did it ! I fixed this issue.
I had to deleted a device role that had an image assignment
I was on pgAdmin on the database and i execute this request on this table "netbox_topology_views_roleimage"

SELECT *,*
FROM netbox_topology_views_roleimage;

![image](https://github.com/mattieserver/netbox-topology-views/assets/107417012/682bad05-e211-482e-9ea7-d6779f493e88)

and then i compared the id with other table "dcim_devicerole"

SELECT id
FROM dcim_devicerole;

![image](https://github.com/mattieserver/netbox-topology-views/assets/107417012/90e66312-471a-42de-9bc9-ac6b32cd6012)

Finally, i delete the id image 

![image](https://github.com/mattieserver/netbox-topology-views/assets/107417012/5559af64-aa24-43ca-949f-e8b07e5b86a1)

And then everything is ok : 

![image](https://github.com/mattieserver/netbox-topology-views/assets/107417012/428aecca-3679-43c9-871a-400879b5f00b)


Thanks a lot for your feedback,
Max
